src/basic/time-util.c

```diff
@@ -55,6 +56,20 @@ usec_t now(clockid_t clock_id) {
         return timespec_load(&ts);
 }
 
+usec_t now_reproducible(clockid_t clock_id) {
```

Comment:
Not a big matter, but I'd like to rename this `epoch_or_now()` or so. "reproducible" here sounds strange for me.